R/flatten_data.R

```diff
@@ -1,7 +1,42 @@
+#' determine data type
+#' 
+#' @param data (list or tbl_df, tbl, data.frame) An ecocomDP formatted dataset (list), or a list of ecocomDP tables, a single flat ecocomDP table, or a single flat table created by joinng and spreading ecocomDP tables. 
+#' 
+#' @return (character) A character string indicating data type. One of: 'list_of_tables', 'datset_old', 
+#' 'dataset', or 'flat_table'
+#' 
+#' @details Internal function for detecting datatype. Not exported
+detect_data_type <- function(data){
+  
+  table_names <- c(
+    "location",
+    "location_ancillary",
+    "taxon",
+    "observation",
+    "observation_ancillary",
+    "dataset_summary")
+  
+  if((class(data)=="list") && any(table_names %in% names(data))){
+    return("list_of_tables")
+    
+  }else if((class(data)=="list") && (length(data)==1) && ("tables" %in% names(data[[1]]))){
+    return("dataset_old")
+    
+  }else if((class(data)=="list") && ("tables" %in% names(data))){
+    return("dataset")
+    
+  }else if(any(class(data) %in% c("data.frame", "tbl_df", "tbl"))){
+    return("flat_table")
+  }else{
+    stop("Not a recognizable data type")
+  }
+}
```

Comment:
@sokole Two other use cases to handle here:
1. A list of old datasets (e.g. `(class(data) == "list") & (length(data) > 1) & ("tables" %in% names(data[[1]]))`)
2. A list of new datasets (e.g. `(class(data) == "list") & (length(data) > 1) & ("tables" %in% names(data))`)

Reply:
@clnsmth can `flatten_data` handle those two cases? I thought it could only flatten one dataset at a time.

Reply:
You're right @sokole, `flatten_data()` shouldn't perform the iteration, but  should `detect_data_type()' should identify the input as a list of datasets new/old, which can be used to throw an error or warning when the user is doing this.